src/UnigramWordPredictor.java

```diff
@@ -49,9 +50,31 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
    * @param scanner the Scanner to read the training text from
    */
   public void train(Scanner scanner) {
+    neighborMap = new HashMap<String,List<String>>();;
     List<String> trainingWords = tokenizer.tokenize(scanner);
+    
+    for(int i = 0; i <= trainingWords.size() - 2; i++){
+      List<String> valueList = new ArrayList<>();
+      String currentWord = trainingWords.get(i);
+      String nextWord = trainingWords.get(i + 1);
 
-    // TODO: Convert the trainingWords into neighborMap here
+      // System.out.println(trainingWords.get(i));
+      // if token is already a key
+      if (neighborMap.containsKey(currentWord)){
+        // add next token to value list
+        valueList = neighborMap.get(currentWord);
+        valueList.add(nextWord);
+        neighborMap.put(currentWord, valueList);
+      }
+
+      // if token isn't a key yet
+      else{
+        // make token a key with value list containing next token
+        valueList.add(nextWord);
+        neighborMap.put(currentWord, valueList);
+      } 
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?